lightning_examples/warp-drive/multi_agent_rl.py

```diff
@@ -0,0 +1,367 @@
+                episode_states["loc_x"][:1, idx] / env.grid_length,
+                episode_states["loc_y"][:1, idx] / env.grid_length,
+                0,
+                color=tagger_color,
+                marker="o",
+                markersize=10,
+            )[0]
+        else:  # runners
+            lines[idx] = ax.plot3D(
+                episode_states["loc_x"][:1, idx] / env.grid_length,
+                episode_states["loc_y"][:1, idx] / env.grid_length,
+                [0],
+                color=runner_color,
+                marker="o",
+                markersize=5,
+            )[0]
+
+    init_num_runners = env.num_agents - env.num_taggers
+
+    def _get_label(timestep, n_runners_alive, init_n_runners):
+        line1 = "Continuous Tag\n"
+        line2 = "Time Step:".ljust(14) + f"{timestep:4.0f}\n"
+        frac_runners_alive = n_runners_alive / init_n_runners
+        pct_runners_alive = f"{n_runners_alive:4} ({frac_runners_alive * 100:.0f}%)"
+        line3 = "Runners Left:".ljust(14) + pct_runners_alive
+        return line1 + line2 + line3
+
+    label = ax.text(
+        0,
+        0,
+        0.02,
+        _get_label(0, init_num_runners, init_num_runners).lower(),
+    )
+
+    label.set_fontsize(14)
+    label.set_fontweight("normal")
+    label.set_color("#666666")
+
+    def animate(i):
+        for idx, line in enumerate(lines):
+            line.set_data_3d(
+                episode_states["loc_x"][i : i + 1, idx] / env.grid_length,
+                episode_states["loc_y"][i : i + 1, idx] / env.grid_length,
+                np.zeros(1),
+            )
+
+            still_in_game = episode_states["still_in_the_game"][i, idx]
+
+            if still_in_game:
+                pass
+            else:
+                line.set_color(runner_not_in_game_color)
+                line.set_marker("")
+
+        n_runners_alive = episode_states["still_in_the_game"][i].sum() - env.num_taggers
+        label.set_text(_get_label(i, n_runners_alive, init_num_runners).lower())
+
+    ani = animation.FuncAnimation(fig, animate, np.arange(0, env.episode_length + 1), interval=1000.0 / fps)
+    plt.close()
+
+    return ani
+
+
+# %% [markdown]
+# The animation below shows a sample realization of the game episode before training, i.e., with randomly chosen agent actions. The $5$ taggers are marked in pink, while the $100$ blue agents are the runners. Both the taggers and runners move around randomly and about half the runners remain at the end of the episode.
+
+# %%
+
+# anim = generate_tag_env_rollout_animation(wd_module)
+# HTML(anim.to_html5_video())
+
+# %% [markdown]
+# ## Create the Lightning Trainer
+#
+# Next, we create the trainer for training the WarpDrive model. We add the `performance stats` callbacks to the trainer to view the throughput performance of WarpDrive.
+
+# %%
+log_freq = run_config["saving"]["metrics_log_freq"]
+
+# Define callbacks.
+cuda_callback = CudaCallback(module=wd_module)
```

Comment:
```suggestion
cuda_callback = CUDACallback(module=wd_module)
```
Nit: Name it CUDACallback in main repo